media/server/main/include/MediaPipelineServerInternal.h

```diff
@@ -224,6 +222,11 @@ class MediaPipelineServerInternal : public IMediaPipelineServerInternal, public
      */
     std::map<MediaSourceType, std::int32_t> m_attachedSources;
 
+    /**
+     * @brief Map of flags for counter w mediasource
+     */
+    std::map<MediaSourceType, int> m_attachedSourcesForCounter;
```

Comment:
I think this could have a better name, something like m_noAvailableSamplesCounter

And then change the description to something like, 'Map to count the number of NO_AVAILABLE_SAMPLES MediaSourceStatus's per MediaSource'